### PR TITLE
Add consumer bus config

### PR DESCRIPTION
**Pull request by maisiesadler**
- Add ConsumerBus config to IMessagingConfig
- Provide default implementation ConsumerConfig

### CODE REVIEW

File: src/JustSaying/IMessagingConfig.cs [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 {
-    public interface IMessagingConfig : IPublishConfiguration //ToDo: This vs publish config. Clean it up. not good.
+    public interface IMessagingConfig : IPublishConfiguration
     {
         IList<string> Regions { get; }
         Func<string> GetActiveRegion { get; set; }
         IMessageSubjectProvider MessageSubjectProvider { get; set; }
         ITopicNamingConvention TopicNamingConvention { get; set; }
         IQueueNamingConvention QueueNamingConvention { get; set; }
+        IConsumerBusConfig ConsumerBusConfig { get; set; }
```
Comment (gkinsman) [inline src/JustSaying/IMessagingConfig.cs]: I wonder if we should avoid putting `Bus` in here. The concept of a bus is kinda encapsulated in `IJustSayingBus` and doesnt mean much from a config perspective. `IConsumerConfig` perhaps? Thoughts?

File: src/JustSaying/Messaging/Channels/ConsumerBusConfig.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+namespace JustSaying.Messaging.Channels
+{
+    public class ConsumerBusConfig : IConsumerBusConfig
+    {
+        public int BufferSize => 10;
+
+        public int MultiplexerCapacity => 100;
+
+        public SqsPolicyAsync<IList<Amazon.SQS.Model.Message>> SqsPolicy { get; private set; }
+                    = new InnerSqsPolicyAsync<IList<Amazon.SQS.Model.Message>>();
```
Comment (gkinsman) [inline src/JustSaying/Messaging/Channels/ConsumerBusConfig.cs]: Need to rebase this on the base branch? Was this renamed noop?

File: tests/JustSaying.UnitTests/Messaging/Channels/ChannelsTests.cs [hunk 3]
```diff
@@ -33,11 +33,11 @@ public ChannelsTests(ITestOutputHelper testOutputHelper)
         [Fact]
         public async Task QueueCanBeAssignedToOnePump()
         {
-            var sqsQueue = TestQueue("one");
+            var sqsQueue = TestQueue();
             var buffer = CreateMessageReceiveBuffer(sqsQueue);
             IMessageDispatcher dispatcher = TestDispatcher();
             IChannelConsumer consumer = CreateChannelConsumer(dispatcher);
-            IMultiplexer multiplexer = new RoundRobinQueueMultiplexer(NullLoggerFactory.Instance);
+            IMultiplexer multiplexer = CreateMessageRecieveBuffer();
```
Comment (gkinsman) [inline tests/JustSaying.UnitTests/Messaging/Channels/ChannelsTests.cs]: Should be `CreateMultiplexer`?

File: src/JustSaying/IMessagingConfig.cs (hunk 1 above)
Comment (maisiesadler) [reply src/JustSaying/IMessagingConfig.cs]: oo yeah good point, yeah `ConsumerConfig` or `SubscriberConfig` make sense i think

File: src/JustSaying/Messaging/Channels/ConsumerBusConfig.cs (hunk 2 above)
Comment (maisiesadler) [reply src/JustSaying/Messaging/Channels/ConsumerBusConfig.cs]: Looks like i renamed the file but not the class 🙃

**Pull request merged**